web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsWriter.java

```diff
@@ -200,21 +273,36 @@ public long getFailuresWhenOthersSucceeded() {
             return failuresWhenOthersSucceeded;
         }
 
+        /**
+         * How long has it been since the query metric update first failed.
+         *
+         * @return the age
+         */
         public long getAge() {
             return System.currentTimeMillis() - created;
         }
     }
 
     private class MetricProcessor implements Runnable {
+        // noinspection unchecked
+        private Map<String,Long> lastPageMetricMap = new LRUMap(1000);
+        private List<FailureRecord> failedMetrics = new ArrayList<>();
+
+        private LinkedBlockingQueue<QueryMetricHolder> blockingQueue;
+
+        public MetricProcessor(LinkedBlockingQueue<QueryMetricHolder> blockingQueue) {
+            this.blockingQueue = blockingQueue;
+        }
+
         @Override
         public void run() {
-            while (!shuttingDown) {
+            while (!shutDownMetricProcessors) {
```

Comment:
So this loop is going to run forever, as fast as the processor will allow it to run, whether there are metrics or not?  This concerns me - it's going to eat up a lot of CPU, isn't it?  At the bare minimum, we should Thread.sleep for a bit if there are no metrics in the queue.  

Alternatively, you could `poll` the queue, and give a timeout for the poll call.  If there's a metric available to write, it will do so immediately, if not, it will wait the specified amount of time.  Calling `take()` would wait indefinitely until a metric is available, which might be even better in this case.

Reply:
That loop calls getMetricsFromQueue which calls poll with a timeout and loops until batchSize or maxLatency  is reached.